homework/index.js

```diff
@@ -30,14 +30,77 @@
     return elem;
   }
 
+  function renderRepository(leftContainer, repository) {
+    leftContainer.innerHTML = '';
+    const table = createAndAppend('table', leftContainer);
+    const tBody = createAndAppend('tBody', table);
+    const trRepository = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trRepository, { text: 'Repository: ' });
+    createAndAppend('a', trRepository, {
+      text: repository.name,
+      href: repository.html_url,
+    });
+    const trDescription = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trDescription, { text: 'Description: ' });
+    createAndAppend('td', trDescription, { text: repository.description });
+    const trForks = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trForks, { text: 'Forks: ' });
+    createAndAppend('td', trForks, { text: repository.forks });
+    const trUpdated = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trUpdated, { text: 'Updated: ' });
+    createAndAppend('td', trUpdated, { text: repository.updated_at.toLocaleString() });
+  }
+
+  function renderContributors(rightContainer, repository, root) {
+    rightContainer.innerHTML = '';
+    const contributorsURL = repository.contributors_url;
+    createAndAppend('p', rightContainer, { text: 'Contributions', id: 'contTitle' });
+
+    fetchJSON(contributorsURL, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      }
+
+      contributors.forEach(contributor => {
+        createAndAppend('ul', rightContainer, { class: 'ul' });
+        const li = createAndAppend('li', rightContainer, { class: 'li' });
+        createAndAppend('img', li, { src: contributor.avatar_url, class: 'contImg' });
+        createAndAppend('a', li, {
+          text: contributor.login,
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contLink',
+        });
+        createAndAppend('div', li, { text: contributor.contributions, class: 'contNumber' });
+      });
```

Comment:
You are creating invalid HTML for the contributor, for example:

```html
<ul class="ul"></ul>
<li class="li">
  <img src="https://avatars2.githubusercontent.com/u/7113309?v=4" class="contImg" />
  <a href="https://github.com/mkruijt" target="_blank" class="contLink">mkruijt</a>
  <div class="contNumber">246</div>
</li>
```

You should generate one `ul` only for the who list of contributors. Then each contributor should have an `li` as its main element, appended as a child to the single `ul` representing the contributors list and inside the `li` the detail content for the contributor.

Reply:
Hello @remarcmij code has been improved based on your comment.
Thanks.